source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+
+              AssertThrow(spherical_representative_point[2] >= gm->south_latitude() &&
+                          spherical_representative_point[2] <= gm->north_latitude(),
+                          ExcMessage("Latitude of representative point outside chunk domain."));
+            }
+
+          // set outer radius
+          spherical_representative_point[0] = gm->outer_radius();
+        }
+//      TODO There will be a LayeredChunk soon :)
+//      else if (const GeometryModel::TwoMergedChunks<dim> *gm = dynamic_cast<const GeometryModel::TwoMergedChunks<dim>*> (&this->get_geometry_model()))
+//        {
+//          spherical_representative_point[1] *= degrees_to_radians;
+//
+//          AssertThrow(spherical_representative_point[1] >= gm->west_longitude() &&
+//                      spherical_representative_point[1] <= gm->east_longitude(),
+//                      ExcMessage("Longitude of representative point outside chunk domain."));
+//
+//          if (dim ==3)
+//            {
+//              spherical_representative_point[2] *= degrees_to_radians;
+//
+//              AssertThrow(spherical_representative_point[2] >= gm->south_latitude() &&
+//                          spherical_representative_point[2] <= gm->north_latitude(),
+//                          ExcMessage("Latitude of representative point outside chunk domain."));
+//            }
+            // set outer radius
+//          spherical_representative_point[0] = gm->outer_radius();
+//        }
+      else if (const GeometryModel::EllipsoidalChunk<dim> *gm = dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*> (&this->get_geometry_model()))
+        {
+          // NB: the semi major axis a is only equal to the radius at all surface points if the eccentricity is zero.
+
+          // TODO min/max angles are not available from geometry plugin
+//          AssertThrow(//spherical_representative_point[0] >= gm->inner_radius() &&
+//                      spherical_representative_point[0] <= gm->get_semi_major_axis_a() &&
+//                      spherical_representative_point[1] >= gm->minimum_longitude() &&
+//                      spherical_representative_point[1] <= gm->maximum_longitude(),
+//                      ExcMessage("Longitude of representative point outside chunk domain."));
+
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          if (dim ==3)
+            {
+
+//              AssertThrow(spherical_representative_point[2] >= gm->minimum_latitude() &&
+//                          spherical_representative_point[2] <= gm->maximum_latitude(),
+//                          ExcMessage("Representative point outside chunk domain."));
+
+              spherical_representative_point[2] *= degrees_to_radians;
+            }
+
+          // TODO this is only correct for eccentricity zero
+          // or ask for the correct radius from user
+          spherical_representative_point[0] = gm->get_semi_major_axis_a();
```

Comment:
I guess we still have to think about how to do this for the EllipsoidalChunk. Is there an easy way to compute the outer radius for a given point when using the corners/eccentricity/...? Otherwise we could also add a function to the EllipsoidalChunk that computes what we want.